content-test/addon/PageWorker.test.js

```diff
@@ -0,0 +1,85 @@
+const PageWorker = require("addon/PageWorker");
+const {LOCAL_STORAGE_KEY} = require("common/constants");
+const {ADDON_TO_CONTENT} = require("common/event-constants");
+const redux = require("redux");
+
+// Test reducer that adds action.data to a total
+function reducer(state = 0, action) {
+  return action.type === "add" ? (state + action.number) : state;
+}
+function createStore() {
+  return redux.createStore(reducer, 0);
+}
+
+describe("PageWorker", () => {
+  let pageWorker;
+  let store;
+  beforeEach(() => {
+    store = createStore();
+    pageWorker = new PageWorker({store});
+  });
+  afterEach(() => pageWorker.destroy());
+  it("should create a PageWorker with a store", () => {
+    assert.equal(pageWorker._store, store);
+    assert.isNull(pageWorker._page);
+    assert.isNull(pageWorker._unsubscribe);
+  });
+  it("should throw if you try to create a PageWorker without a store", () => {
+    assert.throws(() => {
+      assert.ok(new PageWorker());
+    }, "options.store is required");
+  });
+  describe("#connect", () => {
+    it("should create a Page instance and ._unsubscribe", () => {
+      pageWorker.connect();
+      assert.isObject(pageWorker._page);
+      assert.isFunction(pageWorker._unsubscribe);
+    });
+    it("should subscribe to the store with ._onDispatch", () => {
+      sinon.spy(pageWorker, "_onDispatch");
+      pageWorker.connect();
+      store.dispatch({type: "add", number: 42});
+      assert.equal(store.getState(), 42);
+      assert.called(pageWorker._onDispatch);
```

Comment:
nit: this test is using side effects to make assertions. judging by the description you want to:
1. `stub` (so no side effects) the subscribe method of the store 
2. call `connect`
3. assert that `store.subscribe` was `calledOnce` and `calledWithExactly` `_onDispatch`